src/internal/transaction.ts

```diff
@@ -118,8 +133,16 @@ export async function waitForTransaction(args: {
       break;
     }
     try {
-      // eslint-disable-next-line no-await-in-loop
-      lastTxn = await getTransactionByHash({ aptosConfig, transactionHash });
+      if (!isFirstTry && !longWaitOnce) {
+        // We are greedy, hence do a 'long wait' once to shave off latencies resulting from polling with backoff
+        longWaitOnce = true;
+        // eslint-disable-next-line no-await-in-loop
+        lastTxn = await longWaitForTransaction({ aptosConfig, transactionHash });
+      } else {
+        isFirstTry = false;
```

Comment:
These two variables are the same right?

Maybe just make isFirstTry = false when you first enter the first section
```suggestion
      if (!triedOnce) {
        // We are greedy, hence do a 'long wait' once to shave off latencies resulting from polling with backoff
        triedOnce = true;
        // eslint-disable-next-line no-await-in-loop
        lastTxn = await longWaitForTransaction({ aptosConfig, transactionHash });
      } else {
```

Reply:
No, the logic we want is:
Try the polling way in the first attempt hoping there is no need for a wait.
If that doesn't work try to do long wait for 1s.
If the long wait also is not enough, then fallback to polling

Reply:
Okay so:
1. Short wait
2. Long wait
3. Short wait loop

Reply:
Yup